synapseclient/models/agent.py

```diff
@@ -0,0 +1,629 @@
+
+        my_agent.prompt(
+            prompt="Hello",
+            enable_trace=True,
+            print_response=True,
+            synapse_client=syn,
+        )
+
+        my_second_session = my_agent.start_session(synapse_client=syn)
+        my_agent.prompt(
+            prompt="Hello again",
+            enable_trace=True,
+            print_response=True,
+            session=my_second_session,
+            synapse_client=syn,
+        )
+    """
+
+    cloud_agent_id: Optional[str] = None
+    """The unique ID of the agent in the cloud provider."""
+
+    cloud_alias_id: Optional[str] = None
+    """The alias ID of the agent in the cloud provider.
+        Defaults to 'TSTALIASID' in the Synapse API.
+    """
+
+    registration_id: Optional[int] = None
+    """The ID number of the agent assigned by Synapse."""
+
+    registered_on: Optional[datetime] = None
+    """The date the agent was registered."""
+
+    type: Optional[AgentType] = None
+    """The type of agent. One of either BASELINE or CUSTOM."""
+
+    sessions: Dict[str, AgentSession] = field(default_factory=dict)
+    """A dictionary of AgentSession objects, keyed by session ID."""
+
+    current_session: Optional[AgentSession] = None
+    """The current session. Prompts will be sent to this session by default."""
+
+    def fill_from_dict(self, agent_registration: Dict[str, str]) -> "Agent":
+        """
+        Converts a response from the REST API into this dataclass.
+
+        Arguments:
+            agent_registration: The response from the REST API.
+
+        Returns:
+            The Agent object.
+        """
+        self.cloud_agent_id = agent_registration.get("awsAgentId", None)
+        self.cloud_alias_id = agent_registration.get("awsAliasId", None)
+        self.registration_id = agent_registration.get("agentRegistrationId", None)
+        self.registered_on = agent_registration.get("registeredOn", None)
+        self.type = agent_registration.get("type", None)
+        return self
+
+    @otel_trace_method(
+        method_to_trace_name=lambda self, **kwargs: f"Register_Agent: {self.registration_id}"
+    )
+    async def register_async(
+        self, *, synapse_client: Optional[Synapse] = None
+    ) -> "Agent":
+        """Registers an agent with the Synapse API.
+        If agent already exists, it will be retrieved.
+
+        Arguments:
+            synapse_client: If not passed in and caching was not disabled by
+                    `Synapse.allow_client_caching(False)` this will use the last created
+                    instance from the Synapse class constructor.
+
+        Returns:
+            The registered or existing Agent object.
+        """
+        syn = Synapse.get_client(synapse_client=synapse_client)
+        agent_response = await register_agent(
+            cloud_agent_id=self.cloud_agent_id,
+            cloud_alias_id=self.cloud_alias_id,
+            synapse_client=syn,
+        )
```

Comment:
You can drop the extra call to `get_client` since you're just passing along whatever was given